### PR TITLE
Azure Monitor: Fix broken links in readme

### PR DESCRIPTION
**Pull request by srnagar**
Fixes https://github.com/Azure/azure-sdk-for-java/issues/34210

### CODE REVIEW

File: sdk/monitor/azure-monitor-ingestion/README.md [hunk 1]
```diff
@@ -30,11 +64,12 @@ that you create in Log Analytics workspace. You can even extend the schema of bu
 ### Create the client
 
 An authenticated client is required to upload logs to Azure Monitor. The library includes both synchronous and asynchronous forms of the clients. To authenticate, 
-the following examples use `DefaultAzureCredentialBuilder` from the [com.azure:azure-identity](https://search.maven.org/artifact/com.azure/azure-identity) package.
+the following examples use `DefaultAzureCredentialBuilder` from the [com.azure:azure-identity](https://central.sonatype.com/artifact/com.azure/azure-identity/1.9.0-beta.2/versions) package.
```
Comment (scottaddie) [inline sdk/monitor/azure-monitor-ingestion/README.md:67]: Is there a way we can avoid linking to a specific version of the package?

Comment (srnagar) [reply sdk/monitor/azure-monitor-ingestion/README.md:67]: Unfortunately, the new maven website doesn't allow us to point to a list of versions without using one of the version numbers in the link.

Comment (JonathanGiles) [reply sdk/monitor/azure-monitor-ingestion/README.md:67]: Yeah, we are going to need tooling to keep these versions updated, but unfortunately Maven Central doesn't have any convenience URLs.

File: sdk/monitor/azure-monitor-query/README.md [hunk 2]
```diff
@@ -224,6 +225,23 @@ LogsQueryResult / LogsBatchQueryResult
         |---type
 ```
 
+#### Logs query by resource id
+The `LogsQueryClient` supports querying logs using a workspace id (queryWorkspace methods) or a resource id (queryResource methods).
```
Comment (scottaddie) [inline sdk/monitor/azure-monitor-query/README.md:229]:
```suggestion
The `LogsQueryClient` supports querying logs using a workspace ID (`queryWorkspace` methods) or a resource ID (`queryResource` methods).
```

File: sdk/monitor/azure-monitor-query/CHANGELOG.md [hunk 3]
```diff
@@ -1,15 +1,18 @@
...
 
-## 1.2.0-beta.1 (Unreleased)
+## 1.2.0-beta.1 (2023-04-12)
 
 ### Features Added
 
-### Breaking Changes
-
-### Bugs Fixed
+- Added `queryResource` methods to `LogsQueryClient` to support querying logs using Azure resource id.
```
Comment (scottaddie) [inline sdk/monitor/azure-monitor-query/CHANGELOG.md:7]:
```suggestion
- Added `queryResource` methods to `LogsQueryClient` to support querying logs using an Azure resource ID.
```

File: sdk/monitor/azure-monitor-ingestion/README.md [hunk 4]
```diff
@@ -17,6 +17,40 @@ that you create in Log Analytics workspace. You can even extend the schema of bu
 
 ### Include the package
 
+#### Include the BOM file
+
+Please include the azure-sdk-bom to your project to take dependency on the General Availability (GA) version of the library. In the following snippet, replace the {bom_version_to_target} placeholder with the version number.
```
Comment (scottaddie) [inline sdk/monitor/azure-monitor-ingestion/README.md:22]:
```suggestion
Please include the `azure-sdk-bom` to your project to take a dependency on the latest stable version of the library. In the following snippet, replace the `{bom_version_to_target}` placeholder with the version number.
```

**Review by scottaddie — APPROVED** [review]
While you're updating the README, please also replace all occurrences of "docs.microsoft.com" with "learn.microsoft.com".

**Review by jairmyree — APPROVED** [review]
LGTM

**Pull request merged**